### PR TITLE
Polyval test Assertion Error Fix

### PR DESCRIPTION
**Pull request by shbiswas834**
This PR does the following:

- Tolerance fix for polyval test associated this Jira [SWDEV-471852](https://ontrack-internal.amd.com/browse/SWDEV-471852)
`tests/cupy_tests/lib_tests - TestPolyval::test_polyval - AssertionError `
- Reverting PR #63  since the UT passes without these changes and with the new tolerance
  - Attached logs has all UTs which call `check_func` from PR 63

[cupy_unit_tests_pr_revert.log](https://github.com/user-attachments/files/17595705/cupy_unit_tests_pr_revert.log)

### CODE REVIEW

**Review by lcskrishna — APPROVED** [review]

File: tests/cupy_tests/lib_tests/test_polynomial.py [hunk 1]
```diff
@@ -694,7 +694,7 @@ def test_polyfit_weighted_diff_types(self, xp, dtype1, dtype2, dtype3):
 class TestPolyval(Poly1dTestBase):
 
     @testing.for_all_dtypes()
-    @testing.numpy_cupy_allclose(rtol={numpy.float16: 1e-2, 'default': 1e-3})
+    @testing.numpy_cupy_allclose(rtol={numpy.float16: 1e-2, 'default': 2e-3})
```
Comment (pnunna93) [inline tests/cupy_tests/lib_tests/test_polynomial.py]: This changes tolerance for both rocm and cuda, please add a conditional check for rocm. You can use runtime.is_hip to check the platform.

**Review by pnunna93 — CHANGES_REQUESTED** [review]
@shbiswas834 the PR is changing file permissions, please revert them and add condition for rocm as mentioned below.

File: tests/cupy_tests/lib_tests/test_polynomial.py (hunk 1 above)
Comment (shbiswas834) [reply tests/cupy_tests/lib_tests/test_polynomial.py]: Hey @pnunna93 Added this check. UT passes
[ut_lib.log](https://github.com/user-attachments/files/17635543/ut_lib.log)

Comment (pnunna93) [reply tests/cupy_tests/lib_tests/test_polynomial.py]: Thanks, please fix file permissions and static checks.

**Review by lcskrishna — APPROVED** [review]

**Review by pnunna93 — APPROVED** [review]
LGTM

**Pull request merged**